e2e/test/E2EMsTestBase.cs

```diff
@@ -44,6 +53,20 @@ public void TestCleanup()
             Logger.Event(TestFinishedEventName, extraProperties);
             // As this is not an application that keeps running, explicitly flushing is required to ensure we do not lose any logs.
             Logger.SafeFlush();
+
+            _listener.Dispose();
+        }
+
+        // Dispose artifacts that are in the scope of the entire E2E test suite, and not of each individual test run.
+        // Test run specific artifacts are cleaned up in TestCleanup() above.
+        public void Dispose()
```

Comment:
Instead of following this approach, why don't we add everything needed to the E2E base including - _logAnalyticsClient. They can be initialized on condition. For dispose, you can just check if this is null or not and then call dispose on it.

Reply:
Is there an advantage of doing that? Why not have individual test classes manage their own resources, and shared resources be manages in the base class.
If we keep adding all resources refernce to the base case, it will end up getting all bloated up, and will be difficult to manage.

Reply:
In case another tests needs it, it knows where to get it and then it knows there is a method to call to dispose it.

Reply:
yes, if there is resource required by >1 class, then we can treat is as a shared resource. There is only one test class using logAnalyticsClient, so I don't see the point of bring that into the base test class.